libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -471,6 +490,179 @@ uint64_t VoteManager::roundDeterminedFromVotes(size_t two_t_plus_one) {
   return 0;
 }
 
+void VoteManager::sendRewardPeriodCertVotes(uint64_t reward_period) {
```

Comment:
to discuss on meet:
I got it like reward votes is just previous period votes. Why are we processing in differently?

Reply:
Since finalized a PBFT block only need 2t+1 number of cert votes, that still have 1/3 cert votes may not include in previous period finalized block.
For example:
node1 finalized B1 with cert votes {v1, v2, v3, v4}
node2 finalized B1 with cert votes {v2, v3, v4, v5}
node1 never receive v5 because rebooting(or some other reasons) when gossiping v5. 
When node2 propose a new PBFT block B2 for next period, its reward votes will be {v2, v3, v4, v5}, But node1 doesn't have v5 in finalized B1. node2 has to gossip previous period cert votes before gossip the proposal block B2